src/user/user.service.ts

```diff
@@ -88,62 +98,48 @@ export class UserService {
       }
     }
 
-    const createUserObject = this.userRepository.create({
-      name,
-      email,
-      contactPermission,
-      firebaseUid: firebaseUser.uid,
-      signUpLanguage,
-    });
+    let formattedUserObject: GetUserDto | null = null;
 
     try {
-      const createUserResponse = await this.userRepository.save(createUserObject);
-      // Only assign Partner access code if partner access or partner id is supplied
-      const partnerAccessWithPartner = partnerAccessCode
-        ? await this.partnerAccessService.assignPartnerAccessOnSignup({
-            partnerAccessCode,
-            userId: createUserResponse.id,
-          })
-        : partnerId
-        ? await this.partnerAccessService.assignPartnerAccessOnSignupWithoutCode({
-            partnerId,
-            userId: createUserResponse.id,
-          })
-        : undefined;
-
-      // partner segment is for crisp API
-      const partnerSegment = partnerAccessWithPartner
-        ? partnerAccessWithPartner.partner.name
-        : 'public';
+      if (signUpType === SIGNUP_TYPE.PARTNER_USER_WITHOUT_CODE) {
+        formattedUserObject = await this.createPartnerUserWithoutCode(
+          createUserDto,
+          firebaseUser.uid,
+        );
+      } else if (signUpType === SIGNUP_TYPE.PARTNER_USER_WITH_CODE) {
+        formattedUserObject = await this.createPartnerUserWithCode(createUserDto, firebaseUser.uid);
+      } else {
+        formattedUserObject = await this.createPublicUser(createUserDto, firebaseUser.uid);
```

Comment:
Optional: Could check for the `SIGNUP_TYPE.PUBLIC` and then throw error in the else.

Reply:
I dn't think its possible to be not one of those three types 🤔